chia/daemon/server.py

```diff
@@ -231,74 +230,84 @@ async def incoming_connection(self, request):
                     decoded = json.loads(msg.data)
                     if "data" not in decoded:
                         decoded["data"] = {}
-                    response, sockets_to_use = await self.handle_message(ws, decoded)
+                    response, connections = await self.handle_message(ws, decoded)
+                    service_name = decoded["destination"]
                 except Exception as e:
                     tb = traceback.format_exc()
                     self.log.error(f"Error while handling message: {tb}")
                     error = {"success": False, "error": f"{e}"}
                     response = format_response(decoded, error)
-                    sockets_to_use = []
-                if len(sockets_to_use) > 0:
-                    for socket in sockets_to_use:
-                        try:
-                            await socket.send_str(response)
-                        except Exception as e:
-                            tb = traceback.format_exc()
-                            self.log.error(f"Unexpected exception trying to send to websocket: {e} {tb}")
-                            self.remove_connection(socket)
-                            await socket.close()
+                    service_name = "Unknown"
+                    connections = {ws}  # send error back to the sender
+
+                for connection in connections:
+                    if connection.closed:
+                        # strange, connection is closed, but we still have it in the list
+                        self.remove_connection(connection)
+                        continue
+                    try:
+                        await connection.send_str(response)
+                    except Exception as e:
+                        tb = traceback.format_exc()
+                        self.log.error(f"Unexpected exception trying to send to {service_name} (websocket: {e} {tb})")
+                        self.log.info(f"Closing websocket with {service_name}")
+                        self.remove_connection(connection)
+                        await connection.close()
             else:
-                service_name = "Unknown"
-                if ws in self.remote_address_map:
-                    service_name = self.remote_address_map[ws]
+                service_names = self.find_service_names(ws)
+                if not service_names:
+                    service_names = "Unknown"
```

Comment:
`Union[str, List[str]]` isn't a beautiful hint.  Maybe if we're more explicit?

```suggestion
                service_names = self.find_service_names(ws)
                if len(service_names) > 0:
                    formatted_service_names = ", ".join(name for name in service_names)
                else:
                    formatted_service_names = "unknown service"
```

Reply:
I changed the code a bit - how about just using a list with ["Unknown"] - simple short and only for logging.